tracim/tracim/lib/email_fetcher.py

```diff
@@ -0,0 +1,296 @@
+        return body
+
+    @staticmethod
+    def _parse_html_body(html_body: str):
+        soup = BeautifulSoup(html_body)
+        config = BS_HTML_BODY_PARSE_CONFIG
+        for tag in soup.findAll():
+            if tag.name.lower() in config['tag_blacklist']:
+                tag.extract()
+            elif 'class' in tag.attrs:
+                for elem in config['class_blacklist']:
+                    if elem in tag.attrs['class']:
+                        tag.extract()
+            elif 'id' in tag.attrs:
+                for elem in config['id_blacklist']:
+                    if elem in tag.attrs['id']:
+                        tag.extract()
+            elif tag.name.lower() in config['tag_whitelist']:
+                attrs = dict(tag.attrs)
+                for attr in attrs:
+                    if attr not in config['attrs_whitelist']:
+                        del tag.attrs[attr]
+            else:
+                tag.unwrap()
+        return str(soup)
+
+    def _get_mime_body_message(self) -> typing.Optional[Message]:
+        # FIXME - G.M - 2017-11-16 - Use stdlib msg.get_body feature for py3.6+
+        # FIXME - G.M - 2017-11-16 - Check support for non-multipart mail
+        part = None
+        # Check for html
+        for part in self._message.walk():
+            ctype = part.get_content_type()
+            cdispo = str(part.get('Content-Disposition'))
+            if ctype == 'text/html' and 'attachment' not in cdispo:
+                return part
+        # check for plain text
+        for part in self._message.walk():
+            ctype = part.get_content_type()
+            cdispo = str(part.get('Content-Disposition'))
+            if ctype == 'text/plain' and 'attachment' not in cdispo:
+                return part
+        return part
+
+    def get_key(self) -> typing.Optional[str]:
+
+        """
+        First try checking special header, them check 'to' header
+        and finally check first(oldest) mail-id of 'references' header
+        """
+        key = None
+        first_ref = self.get_first_ref()
+        to_address = self.get_to_address()
+        special_key = self.get_special_key()
+
+        if special_key:
+            key = special_key
+        if not key and to_address:
+            key = DecodedMail.find_key_from_mail_address(to_address)
+        if not key and first_ref:
+            key = DecodedMail.find_key_from_mail_address(first_ref)
+
+        return key
+
+    @staticmethod
+    def find_key_from_mail_address(mail_address: str) \
+            -> typing.Optional[str]:
+        """ Parse mail_adress-like string
+        to retrieve key.
+
+        :param mail_address: user+key@something like string
+        :return: key
+        """
+        username = mail_address.split('@')[0]
+        username_data = username.split('+')
+        if len(username_data) == 2:
+            key = username_data[1]
+        else:
+            key = None
+        return key
+
```

Comment:
Code easier to read/understand. Instead of :

```
       if len(username_data) == 2:
           key = username_data[1]
       else:
           key = None
       return key
```

write this:

```
       if len(username_data) == 2:
           return username_data[1]
       return None
```